auto-value-moshi/src/main/java/com/ryanharter/auto/value/moshi/AutoValueMoshiExtension.java

```diff
@@ -275,6 +278,15 @@ private TypeSpec createTypeAdapter(ClassName className, TypeName autoValueClassN
       FieldSpec field = entry.getValue();
       names.add(prop.serializedName());
 
+      TypeVariableName usedDeclaredGenericTypeName = null;
+      if (genericTypeNames != null && prop.type instanceof TypeVariableName)
+        usedDeclaredGenericTypeName = (TypeVariableName) prop.type;
+
+      if (prop.type instanceof ParameterizedTypeName && genericTypeNames != null) {
+        usedDeclaredGenericTypeName = (TypeVariableName) ((ParameterizedTypeName) prop.type)
+            .typeArguments.get(0);
```

Comment:
That is correct. The idea was just to introduce the feature, so right now a property with the type `Map<T, U>` will not work as intended.